cluster/store/service.go

```diff
@@ -221,6 +221,7 @@ func (st *Service) Execute(req *cmd.ApplyRequest) error {
 	if st.store.IsLeader() {
 		return st.store.Execute(req)
 	}
+	st.store.WaitForLeader(context.Background(), time.Duration(6*time.Second))
```

Comment:
We don't know when the leader will be available, and in the migration, this could take minutes. I believe it would be safe to not wait and just return an error so the user can retry

Reply:
that will require the client to have a retry on all endpoint, also will cause failures on all our tests pipeline. 
what I suggest we add TODO here and revisit after testing if we found that that the case, the time to wait for leader it could be configured though via a flag also with waitUntilDBready etc.